Fails to build for me:

```
/nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/arch/x86/Makefile:114: stack-protector enabled but compiler support broken
In file included from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/compiler.h:54:0,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/linkage.h:4,
                 from /nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/kernel.h:6,
                 from /tmp/nix-build-mxu11x0-1.3.11+git2017-07-13-3.18.0.drv-0/mxu11x0-cbbb5ec2045939209117cb5fcd6c7c23bcc109ef-src/driver/mxu11x0.c:20:
/nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/include/linux/compiler-gcc.h:107:30: fatal error: linux/compiler-gcc6.h: No such file or directory
 #include gcc_header(__GNUC__)
compilation terminated.
make[5]: *** [/nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/scripts/Makefile.build:264: /tmp/nix-build-mxu11x0-1.3.11+git2017-07-13-3.18.0.drv-0/mxu11x0-cbbb5ec2045939209117cb5fcd6c7c23bcc109ef-src/driver/mxu11x0.o] Error 1
make[4]: *** [/nix/store/48b4c1zjl5b8l2f70q9hav9kyr7gbwga-linux-3.18.0-dev/lib/modules/3.18.0/source/Makefile:1398: _module_/tmp/nix-build-mxu11x0-1.3.11+git2017-07-13-3.18.0.drv-0/mxu11x0-cbbb5ec2045939209117cb5fcd6c7c23bcc109ef-src/driver] Error 2
make[3]: *** [Makefile:145: sub-make] Error 2
make[2]: *** [Makefile:24: __sub-make] Error 2
make[1]: *** [Makefile:35: module] Error 2
make: *** [Makefile:5: driver_make] Error 2
```

[build.txt](https://github.com/NixOS/nixpkgs/files/1259447/build.txt)

There is no gcc6 header in this file, so we might need newer linux headers.

--- a/pkgs/os-specific/linux/mxu11x0/default.nix
+++ b/pkgs/os-specific/linux/mxu11x0/default.nix
@@ -1,16 +1,13 @@
 { stdenv, fetchFromGitHub, kernel }:
 
-# it doesn't compile anymore on 3.14
-assert stdenv.lib.versionAtLeast kernel.version "3.18";
-
 stdenv.mkDerivation {
-  name = "mxu11x0-1.3.11-${kernel.version}";
+  name = "mxu11x0-1.3.11+git2017-07-13-${kernel.version}";
 
   src = fetchFromGitHub {
     owner = "ellysh";
     repo = "mxu11x0";
-    rev = "de54053d6f297785d77aba9e9c880001519ffddf";
-    sha256 = "1zmqanw22pgaj3b3lnciq33w6svm5ngg6g0k5xxwwijixg8ri3lf";
+    rev = "cbbb5ec2045939209117cb5fcd6c7c23bcc109ef";
+    sha256 = "0wf44pnz5aclvg2k1f8ljnwws8hh6191i5h06nz95ijbxhwz63w4";
   };
 
   preBuild = ''
@@ -19,6 +16,7 @@ stdenv.mkDerivation {
     sed -i -e 's|/lib/modules|${kernel.dev}/lib/modules|' driver/mxconf
     sed -i -e 's|/lib/modules|${kernel.dev}/lib/modules|' driver/Makefile
   '';
+  
   installPhase = ''
     install -v -D -m 644 ./driver/mxu11x0.ko "$out/lib/modules/${kernel.modDirVersion}/kernel/drivers/usb/serial/mxu11x0.ko"
     install -v -D -m 644 ./driver/mxu11x0.ko "$out/lib/modules/${kernel.modDirVersion}/misc/mxu11x0.ko"
@@ -36,5 +34,6 @@ stdenv.mkDerivation {
     license = licenses.gpl1;
     maintainers = with maintainers; [ uralbash ];
     platforms = platforms.linux;
+    broken = (versionOlder kernel.version "4.9");
   };
 }

